tests/test_env.py

```diff
@@ -409,38 +428,35 @@ def assert_all_checks(expect_test: Tuple[bool, str] = (True, ""), expect_nonext:
     assert_all_checks()
     create_install_package("test-package-one", version.Version("2.0.0"), [])
     assert_all_checks(
-        expect_test=(False, "Incompatibility between constraint test-package-one~=1.0 and installed version 2.0.0")
+        expect_test=(False, "Incompatibility between constraint test-package-one~=1.0 and installed version 2.0.0"),
+        expect_nonext=(True, error_msg + " test-package-one~=1.0 and installed version 2.0.0"),
     )
 
 
-def test_active_env_check_constraints(caplog, tmpvenv_active: str) -> None:
+def test_active_env_check_constraints(caplog, tmpvenv_active_inherit: str) -> None:
     """
     Verify that the env.ActiveEnv.check() method's constraints parameter is taken into account as expected.
     """
     caplog.set_level(logging.WARNING)
     in_scope: Pattern[str] = re.compile("test-package-.*")
     constraints: List[Requirement] = [Requirement.parse("test-package-one~=1.0")]
 
-    def check_log(version: Optional[version.Version]) -> None:
-        assert f"Incompatibility between constraint test-package-one~=1.0 and installed version {version}" in {
-            rec.message for rec in caplog.records
-        }
-
-    assert env.ActiveEnv.check(in_scope)
+    env.ActiveEnv.check(in_scope)
 
     caplog.clear()
-    assert not env.ActiveEnv.check(in_scope, constraints)
-    check_log(None)
+    with pytest.raises(env.ConflictingRequirements):
+        env.ActiveEnv.check(in_scope, constraints)
 
     caplog.clear()
     create_install_package("test-package-one", version.Version("1.0.0"), [])
-    assert env.ActiveEnv.check(in_scope, constraints)
+    env.ActiveEnv.check(in_scope, constraints)
+    assert "Incompatibility between constraint" not in "caplog.tex"
```

Comment:
```suggestion
    assert "Incompatibility between constraint" not in caplog.text
```

Reply:
I should re-read my changes. sorry for this